compat/mingw.c

```diff
@@ -3009,6 +3009,45 @@ static char *wcstoutfdup_startup(char *buffer, const wchar_t *wcs, size_t len)
 	return memcpy(malloc_startup(len), buffer, len);
 }
 
+static void maybe_redirect_std_handle(const wchar_t *key, DWORD std_id, int fd,
+				      DWORD desired_access, DWORD flags)
+{
+	DWORD create_flag = fd ? OPEN_ALWAYS : OPEN_EXISTING;
+	wchar_t buf[MAX_LONG_PATH];
+	DWORD max = ARRAY_SIZE(buf);
+	HANDLE handle;
+	DWORD ret = GetEnvironmentVariableW(key, buf, max);
+
+	if (!ret || ret >= max)
+		return;
+
+	/* make sure this does not leak into child processes */
+	SetEnvironmentVariableW(key, NULL);
+	if (!wcscmp(buf, L"off")) {
+		handle = GetStdHandle(std_id);
+		if (handle != INVALID_HANDLE_VALUE)
+			CloseHandle(handle);
+		return;
+	}
+	handle = CreateFileW(buf, desired_access, 0, NULL, create_flag,
+			flags, NULL);
+	if (handle != INVALID_HANDLE_VALUE) {
+		int new_fd = _open_osfhandle((intptr_t)handle, O_BINARY);
+		SetStdHandle(std_id, handle);
+		dup2(new_fd, fd);
+	}
+}
+
+static void maybe_redirect_std_handles(void)
```

Comment:
@jeffhostetler I guess you are correct.

Now, I could think of one very inelegant way to solve this: hardcode something like

```
if (!strcmp(key, "GIT_REDIRECT_STDERR")) {
[... get GIT_REDIRECT_STDOUT ...]
[... if it is the same, duplicate GetStdHandle(STD_OUTPUT_HANDLE) and set STD_ERROR_HANDLE accordingly, along with fd = 2 ...]
}
```

Needless to say, there is little chance that upstream would accept anything like this.

OTOH if we special-case the value `2>&1` for `GIT_REDIRECT_STDERR` in such a way, it would make it easier to use, *and* it would make the code more elegant.

What do you think?